Katana/Store.swift

```diff
@@ -18,10 +18,22 @@ public protocol AnyStore: class {
    Dispatches a `Dispatchable` item
    
    - parameter dispatchable: the dispatchable to dispatch
-   - returns: a promise that is resolved when the dispatchable is handled by the store
-  */
+   - returns: a promise parameterized to void that is resolved when the dispatchable is handled by the store
+   */
   @discardableResult
-  func dispatch(_ dispatchable: Dispatchable) -> Promise<Void>
+  func dispatch<T: Dispatchable>(_ dispatchable: T) -> Promise<Void>
```

Comment:
Why do we need this one?

Reply:
I guess we can change this to
`func dispatch<T: StateUpdater>(_ dispatchable: T) -> Promise<Void>`
and this should resolve also the comment below